src/Emanote/Pandoc/BuiltinFiltersSpec.hs

```diff
@@ -0,0 +1,48 @@
+module Emanote.Pandoc.BuiltinFiltersSpec where
+
+import Emanote.Pandoc.BuiltinFilters
+import Emanote.Pandoc.Markdown.Parser (parseMarkdown)
+import Hedgehog
+import Relude
+import Test.Hspec
+import Test.Hspec.Hedgehog (hedgehog)
+import Text.Pandoc.Definition (Inline (..))
+import Text.Pandoc.Walk (query)
+
+spec :: Spec
+spec =
+  do
+    describe "setExternalLinkicon" $ do
+      let run =
+            (>>=) . evalEither . fmap (query extractAttr . preparePandoc . snd) . parseMarkdown ""
```

Comment:
I have a hard time understanding this code. Could you rewrite it verbosely without using function composition and the like?

Reply:
I hope it looks better now.